### PR TITLE
19 add card management system

### LINKED ISSUE #19: Add Card Management System

### PR DESCRIPTION
**Pull request by ezeBalsamo**
Closes #19

### CODE REVIEW

File: source/YugiAcademy/YuGiOh/cards/card_management_system.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+def raise_not_found_card_named(name, card_type):
+    raise SystemRestrictionInfringed(f'There is no {card_type} named {name}.')
+
+
+def card_named(name, card_type, repository, if_found, if_none):
+    try:
+        card = repository.get(name=name)
+        return card if if_found is None else if_found(card)
+    except ObjectDoesNotExist:
+        raise_not_found_card_named(name, card_type) if if_none is None else if_none()
```
Comment (NicolasGuidoBarbeito) [inline source/YugiAcademy/YuGiOh/cards/card_management_system.py:24]: I don´t understand what you are doing. if if_none -> is None else if_none().
It´s not descriptive. It's a "trabalenguas"

File: source/YugiAcademy/YuGiOh/cards/spell_card.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @classmethod
+    def named(cls, name: str, type: str, description: str):
+        enforce_not_blank(name, "Name")
+        enforce_not_blank(type, "Type")
+        enforce_not_blank(description, "Description")
+
+        return cls(name=name, type=type, description=description)
+
+    def synchronize_with(self, spell_card):
```
Comment (NicolasGuidoBarbeito) [inline source/YugiAcademy/YuGiOh/cards/spell_card.py:22]: I think that synchronize it's not representative. Upgrade or replace it's better. Or something like that. Synchronize seems to be a Synchronization with a future link between objects.

**Review by NicolasGuidoBarbeito — COMMENTED** [review]
See comments! Congratulations on your engagement

File: source/YugiAcademy/YuGiOh/cards/spell_card.py (hunk 2 above)
Comment (ezeBalsamo) [reply source/YugiAcademy/YuGiOh/cards/spell_card.py:22]: Already explained!

File: source/YugiAcademy/YuGiOh/cards/card_management_system.py (hunk 1 above)
Comment (ezeBalsamo) [reply source/YugiAcademy/YuGiOh/cards/card_management_system.py:24]: Already explained!

**Review by NicolasGuidoBarbeito — APPROVED** [review]
Excelent work crack!

**Pull request merged**